pandas/core/arrays/string_arrow.py

```diff
@@ -433,3 +443,114 @@ def _str_rstrip(self, to_strip=None):
+                na_value = np.nan
+            else:
+                na_value = False
+            try:
+                result = lib.map_infer_mask(
+                    arr,
+                    f,
+                    mask.view("uint8"),
+                    convert=False,
+                    na_value=na_value,
+                    dtype=np.dtype(dtype),  # type: ignore[arg-type]
+                )
+                return result
+
+            except ValueError:
+                result = lib.map_infer_mask(
+                    arr,
+                    f,
+                    mask.view("uint8"),
+                    convert=False,
+                    na_value=na_value,
+                )
+                if convert and result.dtype == object:
+                    result = lib.maybe_convert_objects(result)
+                return result
+
+        elif is_string_dtype(dtype) and not is_object_dtype(dtype):
+            # i.e. StringDtype
+            result = lib.map_infer_mask(
+                arr, f, mask.view("uint8"), convert=False, na_value=na_value
+            )
+            result = pa.array(result, mask=mask, type=pa.string(), from_pandas=True)
+            return type(self)(result)
+        else:
+            # This is when the result type is object. We reach this when
+            # -> We know the result type is truly object (e.g. .encode returns bytes
+            #    or .findall returns a list).
+            # -> We don't know the result type. E.g. `.get` can return anything.
+            return lib.map_infer_mask(arr, f, mask.view("uint8"))
+
+    def _convert_int_dtype(self, result):
+        if result.dtype == np.int32:
+            result = result.astype(np.int64)
+        return result
+
+    def _str_count(self, pat: str, flags: int = 0):
+        if flags:
+            return super()._str_count(pat, flags)
+        result = pc.count_substring_regex(self._pa_array, pat).to_numpy()
+        return self._convert_int_dtype(result)
+
+    def _str_len(self):
+        result = pc.utf8_length(self._pa_array).to_numpy()
+        return self._convert_int_dtype(result)
+
+    def _str_find(self, sub: str, start: int = 0, end: int | None = None):
+        if start != 0 and end is not None:
+            slices = pc.utf8_slice_codeunits(self._pa_array, start, stop=end)
+            result = pc.find_substring(slices, sub)
+            not_found = pc.equal(result, -1)
+            offset_result = pc.add(result, end - start)
+            result = pc.if_else(not_found, result, offset_result)
+        elif start == 0 and end is None:
+            slices = self._pa_array
+            result = pc.find_substring(slices, sub)
+        else:
+            return super()._str_find(sub, start, end)
+        return self._convert_int_dtype(result.to_numpy())
+
+    def _cmp_method(self, other, op):
+        result = super()._cmp_method(other, op)
+        return result.to_numpy(na_value=False).astype(np.bool_)
```

Comment:
```suggestion
        return result.to_numpy(np.bool_, na_value=False)
```

I think that should avoid an intermediate conversion to object dtype

Reply:
No, both statement are equivalent. Generally, we shouldn't get object here anyway, but better enforce this

Reply:
> No, both statement are equivalent. Generally, we shouldn't get object here anyway

It actually does matter, without specifying a dtype we convert to object at the moment:

```
In [71]: pd.array([True, False, None], dtype="bool[pyarrow]").to_numpy(na_value=False)
Out[71]: array([True, False, False], dtype=object)
```

Although we should probably fix that .., to ensure this works out of the box when specifying a `na_value` that matches the native numpy dtype. 

(anyway, for this PR you already changed it)

Reply:
Yeah but it's equivalent to

``` 
result.to_numpy(na_value=False).astype(np.bool_)
```

the Current version just does this internally, but it has the same performance drawbacks

Reply:
I think when there are missing values and you are not converting to object dtype, we internally never convert to object dtype:

https://github.com/pandas-dev/pandas/blob/b4b96fb30d21b77d2a7ea67aa4906b69b7e62147/pandas/core/arrays/arrow/array.py#L1273-L1278

We first fill the missing values, and then use pyarrow's to_numpy, which for bools will directly give a bool numpy array